source/type/pt_data.cpp

```diff
@@ -35,6 +35,98 @@ www.navitia.io
 
 namespace navitia { namespace type {
 
+type::Network* PT_Data::get_or_create_network(const std::string& uri, const std::string& name, int sort) {
+    const auto it = networks_map.find(uri);
+    if (it != networks_map.end()) {
+        return it->second;
+    }
+
+    nt::Network* network = new nt::Network();
+    network->uri = uri;
+    network->name = name;
+    network->sort = sort;
+
+    network->idx = networks.size();
+    networks.push_back(network);
+    networks_map[uri] = network;
+
+    return network;
+}
+
+type::CommercialMode* PT_Data::get_or_create_commercial_mode(const std::string& uri, const std::string& name) {
+    const auto it = commercial_modes_map.find(uri);
+    if (it != commercial_modes_map.end()) {
+        return it->second;
+    }
+
+    nt::CommercialMode* mode = new nt::CommercialMode();
+    mode->uri = uri;
+    mode->name = name;
+
+    mode->idx = commercial_modes.size();
+    commercial_modes.push_back(mode);
+    commercial_modes_map[uri] = mode;
+
+    return mode;
+}
+
+type::Line* PT_Data::get_or_create_line(const std::string& uri,
+                                        const std::string& name,
+                                        type::Network* network,
+                                        type::CommercialMode* commercial_mode,
+                                        int sort,
+                                        const std::string& color,
+                                        const std::string& text_color) {
+    const auto it = lines_map.find(uri);
+    if (it != lines_map.end()) {
+        return it->second;
+    }
+
+    nt::Line* line = new nt::Line();
+    line->uri = uri;
+    line->name = name;
+    line->sort = sort;
+    line->color = color;
+    line->text_color = text_color;
```

Comment:
have to fill ?
- std::vector<Route*> route_list;
- std::vector<PhysicalMode*> physical_mode_list;
- std::vector<Calendar*> calendar_list;
- std::vector<LineGroup*> line_group_list;

Reply:
We will have to do it, agreed.

But these are filled with object initially linked to the VJ.
So the plan is to trigger build_relations() at that moment.
Maybe the safest would be to do it in your PR (maybe in a future PR).